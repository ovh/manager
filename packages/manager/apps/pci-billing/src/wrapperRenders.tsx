import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR', currency: 'â‚¬' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
