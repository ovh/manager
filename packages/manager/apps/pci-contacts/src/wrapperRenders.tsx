import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        return Promise.resolve('');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
