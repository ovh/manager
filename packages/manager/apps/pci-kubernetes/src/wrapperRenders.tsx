import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: () => ({
      ovhSubsidiary: 'mocked_ovhSubsidiary',
      currency: { code: 'EUR' },
    }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
