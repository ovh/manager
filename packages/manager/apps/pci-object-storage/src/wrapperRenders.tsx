import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import React from 'react';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);

export const mockShellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: vi.fn().mockReturnValue({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
    getApplication: vi.fn(),
  },
  shell: {
    environment: {
      getRegion: vi.fn().mockReturnValue('EU'),
      getUser: vi.fn().mockReturnValue({}),
      getApplication: vi.fn(),
    },
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

export const wrapperShow = ({ children }: { children: React.ReactNode }) => {
  const router = createMemoryRouter(
    [
      {
        path: '/',
        element: children,
        handle: {
          crumb: () => 'Home',
        },
      },
    ],
    {
      initialEntries: ['/'],
      future: {
        v7_startTransition: true,
        v7_relativeSplatPath: true,
      } as any,
    },
  );

  return (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={mockShellContext as any}>
        <RouterProvider router={router} />
      </ShellContext.Provider>
    </QueryClientProvider>
  );
};

export const wrapperOffsiteReplication = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider value={mockShellContext as any}>
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
