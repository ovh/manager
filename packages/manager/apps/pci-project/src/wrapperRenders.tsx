import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        if (app === 'dedicated' && path === '#/contacts/services') {
          return Promise.resolve('https://contacts-url');
        }
        return Promise.resolve('https://mocked-url');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
