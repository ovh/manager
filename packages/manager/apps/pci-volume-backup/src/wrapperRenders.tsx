import { ReactNode } from 'react';
import { QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';
import queryClient from './queryClient';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn(),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = () => {
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={shellContext}>
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
