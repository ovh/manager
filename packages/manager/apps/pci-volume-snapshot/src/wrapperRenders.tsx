import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

export const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      mutations: {
        retry: false,
      },
      queries: {
        retry: false,
      },
    },
  });
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider
        value={(shellContext as unknown) as ShellContextType}
      >
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
