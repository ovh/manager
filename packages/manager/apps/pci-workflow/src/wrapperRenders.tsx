import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();
export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);
