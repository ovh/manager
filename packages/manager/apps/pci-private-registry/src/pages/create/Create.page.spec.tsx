import { describe, vi } from 'vitest';
import { render } from '@testing-library/react';
import * as pciCommonModule from '@ovh-ux/manager-pci-common';
import * as odsComponentsModule from '@ovhcloud/ods-components/react';
import { UseQueryResult } from '@tanstack/react-query';
import { ResponseAPIError, TProject } from '@ovh-ux/manager-pci-common';
import { OsdsBreadcrumb } from '@ovhcloud/ods-components/react';
import CreatePage from '@/pages/create/Create.page';
import { wrapper } from '@/wrapperRenders';

const compute = () => render(<CreatePage />, { wrapper });

describe('CreatePage', () => {
  vi.mock('@ovhcloud/ods-components/react', async (importOriginal) => {
    const actual = (await importOriginal()) as typeof import('@ovhcloud/ods-components/react');
    return {
      ...actual,
      OsdsBreadcrumb: () => <div data-testid="breadcrumb">Breadcrumb</div>,
    };
  });
  vi.mock('@ovh-ux/manager-react-components', (importOriginal) => ({
    ...importOriginal,
    Headers: () => <div>Headers</div>,
    Notifications: () => <div>Notifications</div>,
    useNotifications: () => ({ push: () => null }),
    useProjectUrl: () => 'projectUrl',
  }));

  vi.mock('@ovh-ux/manager-pci-common', (importOriginal) => ({
    ...importOriginal,
    useProject: () => ({
      data: {
        projectId: 'projectId',
        description: 'projectDescription',
      },
    }),
    useProjectLocalisation: () => ({ region: 'region' }),
    PciDiscoveryBanner: () => null,
    isDiscoveryProject: () => false,
  }));

  vi.mock('react-router-dom', (importOriginal) => ({
    ...importOriginal,
    useParams: () => ({ projectId: 'projectId' }),
    useHref: () => ({ pathname: '' }),
    useNavigate: () => ({ navigate: () => null }),
  }));

  vi.mock('@/pages/create/steps/RegionStep', () => ({
    default: () => <div data-testid="region-step">RegionStep</div>,
  }));

  vi.mock('@/pages/create/steps/NameStep', () => ({
    default: () => <div data-testid="name-step">NameStep</div>,
  }));

  vi.mock('@/pages/create/steps/PlanStep', () => ({
    default: () => <div data-testid="plan-step">PlanStep</div>,
  }));

  vi.mock('react-i18next', () => ({
    useTranslation: vi.fn().mockImplementation(() => ({
      t(key: string) {
        return key;
      },
    })),
  }));

  it('should render', () => {
    const { container } = render(<CreatePage />, { wrapper });
    expect(container).toMatchSnapshot();
  });

  describe('Breadcrumb', () => {
    describe('project is loaded', () => {
      it('should show breadcrumb', async () => {
        vi.spyOn(pciCommonModule, 'useProject').mockReturnValueOnce({
          data: {} as TProject,
        } as UseQueryResult<TProject, ResponseAPIError>);

        const { queryByTestId } = compute();

        expect(queryByTestId('breadcrumb')).toBeInTheDocument();
      });

      it('should show breadcrumb with right props', async () => {
        vi.spyOn(pciCommonModule, 'useProject').mockReturnValueOnce({
          data: {} as TProject,
        } as UseQueryResult<TProject, ResponseAPIError>);

        vi.spyOn(odsComponentsModule, 'OsdsBreadcrumb');

        const props = {
          items: [
            Object({
              href: 'projectUrl',
              label: undefined,
            }),
            {
              href: {
                pathname: '',
              },
              label: 'private_registry_title',
            },
            {
              label: 'private_registry_create',
            },
          ],
        };

        compute();

        expect(OsdsBreadcrumb).toHaveBeenNthCalledWith(1, props, {});
      });
    });

    it('should not show breadcrumb if project is not loaded', () => {
      vi.spyOn(pciCommonModule, 'useProject').mockReturnValue({
        data: undefined,
      } as UseQueryResult<TProject, ResponseAPIError>);

      const { queryByTestId } = compute();

      expect(queryByTestId('breadcrumb')).not.toBeInTheDocument();
    });
  });

  describe.skip('Tracking', () => {
    it('should', () => {});
  });
});
