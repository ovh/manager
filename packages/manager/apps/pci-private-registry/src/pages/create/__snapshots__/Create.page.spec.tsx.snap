// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreatePage > should render 1`] = `
<div>
  <div
    data-testid="breadcrumb"
  >
    Breadcrumb
  </div>
  <div
    class="header mt-8"
  >
    <div>
      Headers
    </div>
  </div>
  <div>
    Notifications
  </div>
  <div
    class="mb-5 sticky top-0 z-50"
  />
  <div
    data-testid="region-step"
  >
    RegionStep
  </div>
  <div
    data-testid="name-step"
  >
    NameStep
  </div>
  <div
    data-testid="plan-step"
  >
    PlanStep
  </div>
</div>
`;
